src/transformers/sagemaker/trainer_sm.py

```diff
@@ -126,12 +132,123 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
             return super().training_step(model, inputs)
 
     def _gather_and_numpify(self, tensors, name):
+        if tensors is None:
+            return
         if self.is_model_parallel_enabled:
             tensors = smp_gather(tensors)
             return nested_numpify(tensors)
         else:
             return super()._gather_and_numpify(tensors, name)
 
+    def save_model(self, output_dir: Optional[str] = None):
+        """
+        Will save the model, so you can reload it using :obj:`from_pretrained()`.
+
+        Will only save from the world_master process (unless in TPUs).
+        """
+        if self.is_model_parallel_enabled:
+            self._save_smp(output_dir)
```

Comment:
We need to use a special save because model parallelism requires to:
- gather the state dict on all processes of d_rank 0 (it triggers a sync across those processes)
- save it only on processes 0